wordpress-mobile/WordPress-FluxC-Android
# Woo/order detail changes

This PR cleans up `OrderStore` and removes the use of `WCOrderModel` when fetching/adding order notes and order shipment trackings. This is needed for the Order Detail refactor to MVVM in woocommerce/woocommerce-android#2844

#### Changes
- Adds new method to `OrderIdentifier` to include the `localOrderId`.  baeb7db
- Removes redundant `WCOrderModel` when fetching order notes + fixes the unit tests. 4a1d6bf
- Removes redundant `WCOrderModel` when fetching order shipment + fixes the unit tests. afeb1f3
- Removed redundant `WCOrderModel` when updating order status + fixes the unit tests. e885049
- Removed redundant `WCOrderModel` when adding order note + fixes the unit tests. 1256f4d
- Removed redundant `WCOrderModel` when adding new shipment tracking + fixes the unit tests. 714b74a
- Removed redundant `WCOrderModel` when adding deleting shipment tracking + fixes the unit tests. 2ea0302
- Added new method to fetch shipment tracking from local db by tracking number. 709334c

#### To test
- Click on `Woo` -> `Orders` -> `Fetch Order Notes` and verify that the order notes are fetched successfully.
- Click on `Woo` -> `Orders` -> `Fetch Order Shipment Trackings` and verify that shipment trackings are fetched successfully.
- Click on `Woo` -> `Orders` -> `Update Latest Order Status`, enter an order status and verify that status is updated successfully.
- Click on `Woo` -> `Orders` -> `Post Order Note`, enter an order note and verify that note is added successfully.
- Click on `Woo` -> `Orders` -> `Add Shipment Tracking To First Order`, enter a tracking number and verify that shipment tracking is added successfully.
- Click on `Woo` -> `Orders` -> `Delete First Shipment Tracking For Order`, enter a tracking number and verify that shipment tracking is deleted successfully.
- Verify that all tests are passing in `WCOrderStoreTest`, `MockedStack_WCOrdersTest`, `ReleaseStack_WCOrderExtTest`, `ReleaseStack_WCOrderTest`, `OrderSqlUtilsTest`.
